src/common/services/XverseService.ts

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable class-methods-use-this */
+import Wallet, { AddressPurpose, BitcoinNetworkType } from 'sats-connect';
+import * as bitcoin from 'bitcoinjs-lib';
+import { WalletService } from '@/common/services/index';
+import * as constants from '@/common/store/constants';
+import { XverseTx } from '@/pegin/middleware/TxBuilder/XverseTxBuilder';
+import {
+  WalletAddress, Tx, SignedTx, BtcAccount, Step,
+} from '../types';
+
+export default class XverseService extends WalletService {
+      satsBtcNetwork: BitcoinNetworkType;
+
+      constructor() {
+        super();
+        switch (this.network) {
+          case constants.BTC_NETWORK_MAINNET:
+            this.satsBtcNetwork = BitcoinNetworkType.Mainnet;
+            break;
+          default:
+            this.satsBtcNetwork = BitcoinNetworkType.Testnet;
+            break;
+        }
+      }
+
+      getAccountAddresses(): Promise<WalletAddress[]> {
+        return new Promise<WalletAddress[]>((resolve, reject) => {
+          const walletAddresses: WalletAddress[] = [];
+          const payload = {
+            purposes: ['ordinals', 'payment'] as AddressPurpose[],
```

Comment:
I think this might not be necessary
```suggestion
            purposes: ['payment'] as AddressPurpose[],
```